schema.graphql

```diff
@@ -1,67 +1,25 @@
-type TX @entity {
-  id: ID!
-  createdAt: String!
-}
-# TOP HATS MAY BE UNNECESSARY
-# Entity that recursively maps an TopHat's entire tree
-type TopHatTree @entity {
-  id: ID!
-  createdAt: String!
-  owner: Bytes!
-  subHats: [HatTree!]
-}
-# Finite TopHat data
-type TopHat @entity {
-  id: ID!
-  createdAt: String!
-  owner: Bytes!
-  subHats: [HatTree!]
-}
-# Entity that recursively maps an Hat's tree
-type HatTree @entity {
-  id: ID!
-  createdAt: String!
-  name: String
-  description: String
-  imageUrl: String
-  # wearer: [WearerTransferReceipt!]
-  subHats: [HatTree!]
-}
-# Finite Hat data
 type Hat @entity {
   id: ID!
-  createdAt: String!
-  # wearerStatus: String!
-  name: String
-  description: String
-  imageUrl: String
-  # wearer: Wearer
-  wearers: [Wearer!]
-}(
-# HatToken vs Hat (Type)
-type HatToken @entity {
-
-}
+  status: Boolean!
+  createdAt: BigInt!
+  wearers: [Wearer!]!
+  details: String!
+  maxSupply: BigInt!
+  eligibility: String!
+  toggle: String!
+  mutable: Boolean!
+  imageUri: String!
+  level: Int!
+  tree: Tree!
+} 
 
 type Wearer @entity {
   id: ID!
-  name: String!
-  age: Int!
-  tx: TX!
-}
-
-type WearerProfile @entity {
-  id: ID!
-  name: String!
-  age: Int!
-  tx: TX!
-  allHats: [Hat!]
+  hats: [Hat!]! @derivedFrom(field: "wearers")
```

Comment:
We'll also want to track

- badStandings (alternatively, create a new HatWearer entity with a standing attribute)
- a history of all hats worn by a given wearer